lib/daemon.py

```diff
@@ -114,6 +117,13 @@ def write_pidfile(pid, pidfile):
     fd.write("%s" % pid)
     fd.close()
 
+    # lock pidfile:
+    try:
+        fd = os.open(pidfile, os.O_RDONLY)
+        fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+        # don't close fd or lock is gone
+    except OSError as e:
+        print("Could not lock pid file: %d (%s)" % (e.errno, e.strerror) , file=sys.stderr)
```

Comment:
Maybe this locking and also be added to the API doc, e.g.: This method writes the PID to the pidfile and locks it while process is running.